src/Microsoft.AspNetCore.Routing/Template/RoutePrecedence.cs

```diff
@@ -92,6 +129,37 @@ private static int ComputeOutboundPrecedenceDigit(TemplateSegment segment)
             }
         }
 
+        private static int ComputeOutboundPrecedenceDigit(RoutePatternPathSegment pathSegment)
+        {
+            if (pathSegment.Parts.Count > 1)
+            {
+                return 4;
+            }
+
+            var part = pathSegment.Parts[0];
+            if (part.IsLiteral)
+            {
+                return 5;
```

Comment:
why 5 (and why 4 above it)

Reply:
See comments elsewhere in this file.

Reply:
Sorry, I should have explained this. None of what's here is new code. It's existing logic that can't be super-re-used without creating a nother overload.